controllers/oscmachine_keypair_controller.go

```diff
@@ -0,0 +1,89 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/services/storage"
+	tag "github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/tag"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/scope"
+)
+
+// getKeyPairResourceId return the keypairName from the resourceMap base on resourceName (tag name + cluster uid)
+func getKeypair(resourceName string, machineScope *scope.MachineScope) (string, error) {
+	keypairRef := machineScope.GetKeypairRef()
+	if keypairName, ok := keypairRef.ResourceMap[resourceName]; ok {
+		return keypairName, nil
+	} else {
+		return "", fmt.Errorf("%s does not exist", resourceName)
+	}
+}
+
+// checkKeypairFormatParameters check Keypair parameters format
+func checkKeypairFormatParameters(machineScope *scope.MachineScope) (string, error) {
+	machineScope.Info("Check Keypair parameters")
+	keypairSpec := machineScope.GetKeypair()
+	//netSpec.SetDefaultValue()
+	keypairRef := machineScope.GetKeypairRef()
+	keyName := keypairSpec.Name + "-" + machineScope.GetUID()
+	keypairTagName, err := tag.ValidateTagNameValue(keyName)
+	if err != nil {
+		return keypairTagName, err
+	}
+	if len(keypairRef.ResourceMap) == 0 {
+		keypairRef.ResourceMap = make(map[string]string)
+	}
+	keypairPublicId := keypairSpec.PublicKey
+	_, err = storage.ValidateSize(keypairPublicId)
```

Comment:
PublicKey should be base64 encoded. (Simple to use in yaml)
And you decoded after.

Reply:
Why size ?

Maybye check enoding publickey is valid. (good format, ...)

Reply:
Ps on api:
https://docs.outscale.com/api#createkeypair
The public key. It must be Base64-encoded.




<br class="Apple-interchange-newline">